core-bundle/tests/Fixtures/Helper/HookHelper.php

```diff
@@ -30,4 +30,27 @@ public function __invoke(...$args)
             '__invoke',
         ];
     }
+
+    public static function registerInsertTagsHook(\Closure $handler): void
+    {
+        $GLOBALS['TL_HOOKS']['replaceInsertTags'][] = [
+            new class($handler) {
+                public function __construct(private readonly \Closure $handler)
+                {
+                }
+
+                // Cannot use ...spread operator because of references
+                public function __invoke(string &$a, bool &$b, string $c, array &$d, array &$e, array $f, int &$g, int|string &$h)
```

Comment:
```suggestion
                public function __invoke(&$a, &$b, $c, &$d, &$e, $f, &$g, &$h)
```
The helper class should not validate the type signature ~~and should pass all parameters by reference~~ so that the test (and not the helper) validates it works correctly IMO.
*EDIT: $c and $f cannot be passed by reference.*